### PR TITLE
fix(mobile): hide keyboard when scroll suggestions

### PR DESCRIPTION
**Pull request by cristianossd**
**Problem:** when scroll suggestions, the keyboard keeps opened as following gif below. This patch fixes this behavior, passing to focus on suggestion items.

Before | After
--- | ---
scroll-before | scroll-after

### CODE REVIEW

File: auto-complete.js [hunk 1]
```diff
@@ -171,14 +174,32 @@
...
+                    }
+
+                    that.sc.focus();
+                }, that.sc);
+            }
+
+            that.blurHandler = function (e) {
+                var isSuggestionsFocused = document.querySelector('.autocomplete-suggestions--focused');
+                if (!isSuggestionsFocused) {
+                    try { var over_sb = document.querySelector('.autocomplete-suggestions:hover'); } catch (e) { var over_sb = 0; }
```
Comment (felipemsantana) [inline auto-complete.js]: `querySelector` returns `null` if element doesn't exist, use `camelCase`.
What does `Sb` mean?
```suggestion
                    var overSb = document.querySelector('.autocomplete-suggestions:hover');
```

File: auto-complete.js [hunk 2]
```diff
@@ -171,14 +174,32 @@
...
+                live('autocomplete-suggestions', 'scroll', function(e) {
+                    if (!that.sc.className.includes(suggestionsFocusClass)) {
+                        that.sc.className += ` ${suggestionsFocusClass}`;
+                    }
+
+                    that.sc.focus();
+                }, that.sc);
+            }
+
+            that.blurHandler = function (e) {
```
Comment (felipemsantana) [inline auto-complete.js]: Is `e` used?

File: auto-complete.js [hunk 3]
```diff
@@ -171,14 +174,32 @@
...
 
-            that.blurHandler = function () {
-                try { var over_sb = document.querySelector('.autocomplete-suggestions:hover'); } catch (e) { var over_sb = 0; }
-                if (!over_sb) {
-                    that.last_val = that.value;
-                    that.sc.style.display = 'none';
-                    setTimeout(function () { that.sc.style.display = 'none'; }, 350); // hide suggestions on fast input
-                } else if (that !== document.activeElement) setTimeout(function () { that.focus(); }, 20);
+            if (o.isMobile) {
+                live('autocomplete-suggestions', 'scroll', function(e) {
```
Comment (felipemsantana) [inline auto-complete.js]: `e` not used
```suggestion
                live('autocomplete-suggestions', 'scroll', function() {
```

File: auto-complete.js [hunk 4]
```diff
@@ -91,10 +92,12 @@
         for (var i = 0; i < elems.length; i++) {
             var that = elems[i];
             var rawData;
+            var suggestionsFocusClass = 'autocomplete-suggestions--focused';
 
             // create suggestions container "sc"
             that.parentPositionDeterminant = getParentPositionDeterminant(that);
             that.sc = document.createElement('div');
+            that.sc.setAttribute('tabindex', '-1');
```
Comment (felipemsantana) [inline auto-complete.js:100]: This is weird, `div` by default can not be focused

File: auto-complete.js (hunk 1 above)
Comment (cristianossd) [reply auto-complete.js]: It's strange, I prefer to not change the current codebase and use another patch to refactor entire code, including standard os JS used: lint, es6, etc

Comment (cristianossd) [reply auto-complete.js]: I think Sb means suggestions body, I really don't know 😭

File: auto-complete.js (hunk 4 above)
Comment (cristianossd) [reply auto-complete.js:100]: Yes, it's not common but acceptable, it was necessary in this case. I prefer to use `-1` to not be a focusable element by user, just using code.

Another approach to remove focus from autocomplete input was to include some input element inside `autocomplete-suggestions` div, but I think it's more weird than first one.

**Pull request merged**